public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -332,10 +332,15 @@ export function buildGridItemForPanel(panel: PanelModel): DashboardGridItem {
 
   if (panel.libraryPanel) {
     vizPanelState.$behaviors!.push(
-      new LibraryPanelBehavior({ uid: panel.libraryPanel.uid, name: panel.libraryPanel.name })
+      new LibraryPanelBehavior({
+        uid: panel.libraryPanel.uid,
+        name: panel.libraryPanel.name,
+        title: panel.title,
+      })
     );
     vizPanelState.pluginId = LibraryPanelBehavior.LOADING_VIZ_PANEL_PLUGIN_ID;
     vizPanelState.$data = undefined;
+    vizPanelState.hoverHeader = !panel.title && !timeOverrideShown && !panel.libraryPanel.name;
```

Comment:
This looks like the wrong fix, the library panel name should not control whether the panel header is shown or not, only the title should